Vagrantfile

```diff
@@ -17,14 +21,22 @@ Vagrant.configure("2") do |config|
   config.vm.define master_name do |master|
     master.vm.network "private_network", ip: "192.168.50.2"
     master.vm.hostname = master_name
+    master.vm.provision :shell, inline: "sed 's/127\.0\.0\.1.*.*/192\.168\.50\.2 k8s-master/' -i /etc/hosts"
```

Comment:
I could also use an explanation here - why the the machine need its own public IP in the hosts file?

Reply:
A some point Vagrant started to mess with CentOS and Ubuntu images by setting 127.0.0.1 to the hostname of the machine. This is terrible as some tools simply stop working. I had a problem getting kublet (or kubeproxy, cant remember) to connect to the master, and this helped.

https://github.com/devopsgroup-io/vagrant-hostmanager/issues/203
https://github.com/devopsgroup-io/vagrant-hostmanager/issues/69

Reply:
But we also add the public IPs to `/etc/hosts` in `install-kubeadm.sh` so the file becomes:

```
vagrant@k8s-master:~$ cat /etc/hosts
127.0.0.1	localhost
127.0.1.1	vagrant.vm	vagrant

# The following lines are desirable for IPv6 capable hosts
::1     localhost ip6-localhost ip6-loopback
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters

192.168.50.2 k8s-master
192.168.50.3 k8s-worker-0
192.168.50.4 k8s-worker-1
```